File: test/day3_test.exs
defmodule Day3Test do
  use ExUnit.Case
  require Logger

  @example File.read!("priv/day3/example.txt") |> String.split("\n")
  @input File.read!("priv/day3/input.txt") |> String.split("\n")

  test "part 1" do
    assert Day3.part1(@example) == 157
    Logger.info("Day 3, Part 1: #{Day3.part1(@input)}")
  end

  test "part 2" do
    assert Day3.part2(@example) == 70
    Logger.info("Day 3, Part 2: #{Day3.part2(@input)}")
  end
end

File: kotlin/aoc/src/test/kotlin/y2024/Day3Test.kt
package y2024

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import utils.readFile

class Day3Test {
    private val day = Day3()
    private val file = "day3"

    @Test
    fun part1() {
        assertEquals(161, day.part1(readFile("/y2024/${file}_sample.txt")))
        assertEquals(161289189, day.part1(readFile("/y2024/${file}.txt")))
    }

    @Test
    fun part2() {
        assertEquals(48, day.part2(readFile("/y2024/${file}_sample2.txt")))
        assertEquals(83595109, day.part2(readFile("/y2024/${file}.txt")))
    }
}
